### PR TITLE
[3.5 | Switch] Separate the Audren driver into its own directory

### PR DESCRIPTION
**Pull request by halotroop2288**
- Separated the Switch audio driver into its own directory, behind a pre-processor flag.
  There may be other audio drivers provided by the LibNX in the future.

### CODE REVIEW

**Comment by halotroop2288** [comment]
This is a pretty disposable PR. I feel like it wouldn't make a huge difference. It did make good practice for learning how to make new directories and files in the engine, though.

File: drivers/audren/audio_driver_audren.cpp
Comment (hokaze) [inline drivers/audren/audio_driver_audren.cpp:1]: Was confused for a whlle with the removal of the os, project_settings, errno and malloc headers and was double-checking the code to see if they definitely went unused and could be removed...

...until I eventually scrolled down and saw they'd been moved to the header file - which I probably should have checked first, in hindsight

Anyway, while this is mostly fine and not something worth holding the PR back over, I do question why these were moved?

Besides making the audren driver stand out from the other platform's audio drivers by moving the includes from the .cpp to the .h, my understanding of C and C++ compilation is that you generally want to put as many of the includes in the .cpp files and the headers should have only the absolute bare minimum for sake of faster compile times

e.g. if a given type definition is used in the header, it'll need including in the .h and not the .cpp, but for includes primarily used for their functions, you'd keep that in the .cpp

Just to be clear: this is me being **incredibly pedantic** and it's an absolutely miniscule niggle - in a file this small it has basicaly zero compile time impact, but keeping headers light and sticking to the same coding style as the other drivers seems like good practice

I have no serious objection here, but felt the need to comment on this for sake of a thorough review.

Comment (halotroop2288) [reply drivers/audren/audio_driver_audren.cpp:1]: > Anyway, while this is mostly fine and not something worth holding the PR back over, I do question why these were moved?

No *real* reason right now. My reasoning for making it its own thing is that LibNX could offer a different audio driver in the future and you'd want a way to switch to the other driver.

This PR is mostly just an experiment. A bit of practice for me.

> my understanding of C and C++ compilation is that you generally want to put as many of the includes in the .cpp files and the headers should have only the absolute bare minimum for sake of faster compile times

I didn't know that. I had to refactor this multiple times because it kept trying to compile this for unsupported platforms like macOS. Moving the includes into the header was just a side effect of that, I thought it might fix the issue somewhere in the middle.

I'm probably going to move the files back into the Switch platform folder to deal with these issues.

**Review by hokaze — APPROVED** [review]
Agree with motivation behind PR, it's better organised this way even if LibNX never adds other audio drivers - and if they do, saves some work in the future

Reviewed all changes, looks good to me!

(I have a longer comment on audio_driver_audren.cpp about moving some the includes to audio_driver_audren.h - but that's purely a matter of opinion on conventions and best practices, it's not a real issue of note nor something I care about enough to insist upon)

File: drivers/audren/audio_driver_audren.cpp
Comment (hokaze) [reply drivers/audren/audio_driver_audren.cpp:1]: No worries, it's a pretty niche thing I only half-remembered from my old Computer Science course's software engineering classes and not something I think would have any real impact here - and even in the worst case it just makes compilation a little bit slower rather than anything too drastic

(I'm also using this review as practice - I've not actually handled anything collaberative via github before, most of my hobby projects are solo, my day job uses a very different system, and collab efforts on stuff like game jams didn't exactly have time for review processes :P)

Comment (halotroop2288) [reply drivers/audren/audio_driver_audren.cpp:1]: Instead of moving the files back, I've added a check for the "switch" platform in the `drivers/SCsub` file before it includes the Audren driver.

Comment (hokaze) [reply drivers/audren/audio_driver_audren.cpp:1]: > hokaze: Anyway, while this is mostly fine and not something worth holding the PR back over, I do question why these were moved?

Clarification: this was me asking about the includes being moved, not the files being put into a new directory for the Audren driver - I fully approve of that and agree with you there

Just taking a look at the last commit now - I know the PR needs a review by someone who actually has write access, but some review is better than nothing, right? “Many eyeballs make all bugs shallow” and all that.

EDIT: or rather, will take another look when the Vita check failing is resolved

**Review by hokaze — APPROVED** [review]
Looks good to me, and has the suggested change from last review applied. Approved.

**Pull request merged**
